### DISCUSSION TITLE
Anyone sucessfully make conjure work with chicken?

### DISCUSSION THREAD

**Discussion by glyh** [Olical/conjure#452] [Q&A]
Here's what I'm trying:
```lua
    use { 'Olical/conjure',
        config = function()
            vim.g['conjure#log#hud#border'] = 'none'
            vim.g['conjure#extract#tree_sitter#enabled'] = true
            vim.g['conjure#mapping#eval_visual'] = 'e'
            vim.g['conjure#mapping#doc_word'] = false

            vim.g['conjure#client#scheme#stdio#command'] = 'chicken-csi'
            vim.g['conjure#client#scheme#stdio#prompt_pattern'] = '#;%d> '
            vim.g['conjure#client#scheme#stdio#value_prefix_pattern'] = false
        end,
    }

```

But it's not working properly

**Comment by russtoku** [comment]
Unfortunately, the interaction with different Scheme interpreters requires tailoring the client code beyond the `conjure#client#scheme#command`, `conjure#client#scheme#prompt_pattern`, and `conjure#client#scheme#value_prefix_pattern`. Currently, the Scheme client only really works well the MIT-Scheme interpreter.

It will take some redesign and  code work to be able to handle different Scheme interpreters as well as fleshing out other client features.

**Comment by seamusriordan** [comment]
@glyh This is what I have used to get it going, though I've not done anything extensive.  For the record I'm working on MacOS

```
  vim.g['conjure#client#scheme#stdio#command'] = 'csi -:c'
  vim.g['conjure#client#scheme#stdio#prompt_pattern'] = '#;%d-> '
  vim.g['conjure#client#scheme#stdio#value_prefix_pattern'] = false
```

The `-:c` is important as it enforced interactive mode (I only know about it from the [conjure documentation](https://github.com/Olical/conjure/wiki/Quick-start:-Scheme-(stdio)) and [this stackoverflow post](https://stackoverflow.com/questions/32193428/trouble-running-chicken-scheme-in-emacs)).  I'm not sure where else it is documented.

Though I'm not sure of your specific issue.
